src/OceanSeaIceModels/InterfaceComputations/interpolate_atmospheric_state.jl

```diff
@@ -106,12 +105,37 @@ function interpolate_atmosphere_state!(interfaces, atmosphere::PrescribedAtmosph
                 auxiliary_time_indexing)
     end
 
-    # Set ocean barotropic pressure forcing
-    barotropic_potential = forcing_barotropic_potential(ocean)
-    ρₒ = coupled_model.interfaces.ocean_properties.reference_density
-    if !isnothing(barotropic_potential)
-        parent(barotropic_potential) .= parent(atmosphere_data.p) ./ ρₒ
+    # barotropic potential in PrescribedAtmosphere?
+    atmosphere_pressure = atmosphere.pressure.data
+    tidal_potential = atmosphere.tidal_potential
+
+    if !isnothing(tidal_potential)
+        tidal_potential_data = tidal_potential.data
+    else
+        tidal_potential_data = nothing
+    end
+
+    # Which forcing is this going to be?
+    u_forcing = ocean.model.forcing.u
+    barotropic_potential = if u_forcing isa BarotropicPotentialForcing
+        u_forcing.potential
+    else
+        n = findfirst(x -> x isa BarotropicPotentialForcing, u_forcing)
+        u_forcing[n].potential
```

Comment:
i wrote a function for this --- why don't you extend that for a tuple if that is indeed what you're trying to do here.

Reply:
I just think its sort of a waste of time to put these if statements in here, its not actuallye asier then doing this legitimately. Here's what's there now:

https://github.com/CliMA/ClimaOcean.jl/blob/2fffee46be453908138b4e1642ac8fdbf8412696/src/OceanSimulations/barotropic_potential_forcing.jl#L27

Reply:
Sounds good, I didn't realize that this function existed